resolwe_bio/processes/support_processors/seqtk_reverse_complement.py

```diff
@@ -0,0 +1,134 @@
+
+        # Reverse complement reads
+        (Cmd['seqtk']['seq', '-r', 'input_reads.fastq.gz'] > f'{name}_complemented.fastq')()
+
+        (Cmd['gzip'][f'{name}_complemented.fastq'])()
+
+        # Save the outputs
+        outputs.fastq = [f'{name}_complemented.fastq.gz']
+
+class ReverseComplementPaired(Process):
+    """Reverse complement FASTQ reads file using Seqtk."""
+
+    slug = 'seqtk-rev-complement-paired'
+    process_type = 'data:reads:fastq:paired:seqtk:comp'
+    name = 'Reverse complement FASTQ (Paired)'
+    requirements = {
+        'expression-engine': 'jinja',
+        'executor': {
+            'docker': {
+                'image' : 'resolwebio/common:1.1.0'
+            },
+        },
+        'resources': {
+            'cores': 1,
+            'memory': 16384,
+        },
+    }
+    data_name = '{{ reads|sample_name|default("?") }}'
+    version = '1.0.0'
+
+
+    class Input:
+        """Input fields to process ReverseComplementPaired."""
+
+        reads = DataField('reads:fastq:paired', label='Reads')
+        select_mate = StringField(
+            label='Select mate',
+            description='Select the which mate should be reverse complemented.',
+            choices=[
+                ('Mate 1', 'Mate 1'),
+                ('Mate 2', 'Mate 2'),
+                ('Both', 'Both')
+            ],
+            default='Mate 1',
+        )
+
+    class Output:
+        """Output fields."""
+
+        fastq = ListField(FileField(), label='Reverse complemented FASTQ file')
+        fastq2 = ListField(FileField(), label='Remaining mate')
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+        name_mate1 = os.path.basename(inputs.reads.fastq[0].path).strip('.fastq.gz')
+        name_mate2 = os.path.basename(inputs.reads.fastq2[0].path).strip('.fastq.gz')
+
+        # Concatenate multilane reads
+        (Cmd['cat'][[reads.path for reads in inputs.reads.fastq]] > f'{name_mate1}_original.fastq.gz')()
+        (Cmd['cat'][[reads.path for reads in inputs.reads.fastq2]] > f'{name_mate2}_original.fastq.gz')()
+
+        if inputs.select_mate == 'Mate 1':
+            # Reverse complement mate 1
+            (Cmd['seqtk']['seq', '-r', f'{name_mate1}_original.fastq.gz'] > f'{name_mate1}_complemented.fastq')()
+            (Cmd['gzip'][f'{name_mate1}_complemented.fastq'])()
+            # Save the outputs
+            outputs.fastq = [f'{name_mate1}_complemented.fastq.gz']
+            outputs.fastq2 = [f'{name_mate2}_original.fastq.gz']
+
+        elif inputs.select_mate == 'Mate 2':
+            # Reverse complement mate 2
+            (Cmd['seqtk']['seq', '-r', f'{name_mate2}_original.fastq.gz'] > f'{name_mate2}_complemented.fastq')()
+            (Cmd['gzip'][f'{name_mate2}_complemented.fastq'])()
+
+            # Save the outputs
+            outputs.fastq = [f'{name_mate1}_original.fastq.gz']
+            outputs.fastq2 = [f'{name_mate2}_complemented.fastq.gz']
+
+        else:
```

Comment:
Personally I would explicitly do three `if` statements, something along the lines of

```
if inputs.select_mate == "Mate 1":
    pass

if inputs.select_mate == "Mate 2":
    pass

if inputs.select_mate == "Both":
    pass
```
... just because it explicitly speaks to choices in the process preamble. I had to take a few minutes and see what the last `else` does (because there's a lot of repeating code, differing in small way between `if/else` clauses) because I wasn't very diligent in reading what choices there are. Once I understood the code I checked the `select_mate` options, it made sense.

Reply:
I disagree. You suggestion implies that all three options could be true at the same time. The nature of if-elif-else is handling three exclusive disjunct options and this is exactly what we are handling with.

Reply:
My reasoning was that this is handled on the `StringField` as I am assuming only one choice can be selected at a time.